src/main/java/heronarts/lx/midi/surface/APC40Mk2.java

```diff
@@ -381,12 +513,19 @@ private void unregister() {
       }
       if (this.device != null) {
         for (int i = 0; i < this.knobs.length; ++i) {
-          if (this.knobs[i] != null) {
+          if (this.knobs[i] == null) {
+            continue;
+          } else if (this.knobs[i] instanceof AggregateParameter) {
+            AggregateParameter ap = (AggregateParameter) this.knobs[i];
```

Comment:
Need to duplicate the Set technique here for seenKnobs... this won't crash the program but it will cause a lot of error log spew about trying to unregister listeners that don't exist. Because if multiple knobs are on the same aggregate parameter, the first will unregister all the listeners, then there will be subsequent redundant calls to removeListener() after it's already gone.